### PR TITLE
Updates for the Core Profile TCK

### LINKED ISSUE #1555: Core Profile: Rename the signature test file to exclude the `-jdk11` suffix
Currently the `core-profile-tck/ca-sigtest` signature test files includes a `sigtest-jdk11` Maven artifact qualifier which ends up being the file name suffix as well. The `-jdk11` part should be removed.

### PR DESCRIPTION
**Pull request by jamezp**
**Fixes Issue**
resolves #1555 

**Describe the change**
Upgrades the Jakarta Annotations API to 3.0 for the core-profile-tck. Also, renames the signature test file to remove the `-jdk11` suffix. 

Migrate the `sigtest-maven-plugin` to the latest vresion.

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @OndroMih @dmatej
@starksm64 @scottmarlow

### CODE REVIEW

**Comment by jamezp** [comment]
There is nothing I can do about the build failure since the build requires a SNAPSHOT build of https://github.com/eclipse-ee4j/jakartaee-tck-tools/tree/master/arquillian

File: el-tck/docs/userguide/src/main/jbake/content/using-examples.inc [hunk 1]
```diff
@@ -22,9 +22,9 @@ sigtest-maven-plugin in the TCK runner as below.
 
 =======================================================================
 <plugin>
-    <groupId>org.netbeans.tools</groupId>
+    <groupId>jakarta.tck</groupId>
     <artifactId>sigtest-maven-plugin</artifactId>
-    <version>1.5</version>
+    <version>2.2</version>
```
Comment (scottmarlow) [inline el-tck/docs/userguide/src/main/jbake/content/using-examples.inc:27]: 2.3 is probably a better release as it has better back compatibility.

Comment (jamezp) [reply el-tck/docs/userguide/src/main/jbake/content/using-examples.inc:27]: The issue with 2.3 is it breaks the `-IgnoreJDKClass` argument which _some_ signature test runners use. With that change you can only pass one `-IgnoreJDKClass` argument, which causes signature tests that pass it more than once to fail. In previous versions passing it for each type you want to ignore was the way to do it.

I think https://github.com/eclipse-ee4j/jakartaee-tck-tools/pull/40 somewhat undoes it, but IIRC it doesn't allow passing parameters to `-IgnoreJDKClass` which also breaks compatibility. 

It feels safer to stick with 2.2 for now.

**Comment by gurunrao** [comment]
Annotation TCK source has been moved to https://github.com/jakartaee/common-annotations-api/tree/master/tck.
IMO, core-profile should download and run Annotation TCK for EE 11, other option is to add documentation in core-profile for user to run annotation TCK.
I believe duplication of signatures in multiple github repository will cause issues for future maintenance/releases.

CC @scottmarlow @scottkurz

**Comment by scottmarlow** [comment]
> Annotation TCK source has been moved to https://github.com/jakartaee/common-annotations-api/tree/master/tck. IMO, core-profile should download and run Annotation TCK for EE 11, other option is to add documentation in core-profile for user to run annotation TCK. I believe duplication of signatures in multiple github repository will cause issues for future maintenance/releases.

I think that the underlying question is should someone running the Core Profile TCK which will include signature tests for all specs included in Core Profile have to run separate signature TCK tests in Annotations/other SPEC TCKs?

The same question for Platform/Web Profile TCK, should users have to run signature tests when running other SPEC TCKs?

**Comment by gurunrao** [comment]
> > Annotation TCK source has been moved to https://github.com/jakartaee/common-annotations-api/tree/master/tck. IMO, core-profile should download and run Annotation TCK for EE 11, other option is to add documentation in core-profile for user to run annotation TCK. I believe duplication of signatures in multiple github repository will cause issues for future maintenance/releases.
> 
> I think that the underlying question is should someone running the Core Profile TCK which will include signature tests for all specs included in Core Profile have to run separate signature TCK tests in Annotations/other SPEC TCKs?
> 
> The same question for Platform/Web Profile TCK, should users have to run signature tests when running other SPEC TCKs?

Annotation TCK has an Signature Test, no other tests are run as part of TCK. Hence my point of non duplication of Signature file [ since the file has already been moved to https://github.com/jakartaee/common-annotations-api/tree/master/tck ].

**Comment by scottmarlow** [comment]
> > > Annotation TCK source has been moved to https://github.com/jakartaee/common-annotations-api/tree/master/tck. IMO, core-profile should download and run Annotation TCK for EE 11, other option is to add documentation in core-profile for user to run annotation TCK. I believe duplication of signatures in multiple github repository will cause issues for future maintenance/releases.
> > 
> > 
> > I think that the underlying question is should someone running the Core Profile TCK which will include signature tests for all specs included in Core Profile have to run separate signature TCK tests in Annotations/other SPEC TCKs?
> > The same question for Platform/Web Profile TCK, should users have to run signature tests when running other SPEC TCKs?
> 
> Annotation TCK has an Signature Test, no other tests are run as part of TCK. Hence my point of non duplication of Signature file [ since the file has already been moved to https://github.com/jakartaee/common-annotations-api/tree/master/tck ].

I agree now  and understand the point better.  

So the Core Profile TCK is currently delegating to the other Specification TCKs to cover signature testing, so  we should do the same with the Annotation TCK which does now cover the signature testing as well.

**Comment by scottmarlow** [comment]
What do you think @jamezp? about ^

**Comment by scottmarlow** [comment]
Just to add more info, for Platform/Web Profile TCKs we always check every SPEC API signature included by the Platform/Web Specification instead of delegating that to the Spec implementation TCKs.

**Comment by jamezp** [comment]
I can understand the desire to not duplicate efforts. I don't really have a strong opinion either way. The one thing we'd need to think about though is how we run it from the Core Profile TCK. While it should be fairly easy from a TCK runner, there's no way to validate they're running the correct TCK. That's the one part I see value in having it be part of the Core Profile TCK.

**Comment by starksm64** [comment]
The user guide is actually a bit opaque on how the common annotations signature tests are run. I guess you would only know to use the core-profile-tck/ca-sigtest by looking at the wf-core-tck-runner example as the user guide itself just says it needs to be run, but does not mention it in the section `Running the Signature Tests` section. It should be added there, and we can just delegate to how to run it to the component spec tck as we do the other component specs in the core profile.

I'm just going to merge this and do some updates the to user guide and dist.

**Pull request merged**